modules/sr_site/manifests/discord_gated_entry.pp

```diff
@@ -0,0 +1,70 @@
+class sr_site::discord_gated_entry (
+  $git_root,
+  $root_dir,
+) {
+  user { 'discord':
+    ensure  => present,
+    comment => 'Discord bot service user',
+    shell   => '/sbin/nologin',
+    gid     => 'users',
+  }
+
+  vcsrepo { $root_dir:
+    ensure    => present,
+    provider  => git,
+    source    => 'https://github.com/srobo/discord-gated-entry',
+    revision  => 'origin/main',
```

Comment:
Who has push access to this repo? There's a privilege escalation route here if those people don't already also have root on the server. Suggest pinning to a specific revision or just `master` (which is the local branch) unless the repo is suitably locked down.

Reply:
It's locked down and requires review, which is better than nothing, but I can pin down to a commit.

It's also worth noting that it won't provide _root_ access, just user access. Which whilst better, isn't quite as impacting.

Reply:
Yes and no -- if there are any install scripts or the like which we have or add here and/or there are vulnerabilities in `git` (as has happened), then this can end up escalating to root fairly easily. `puppet` operates mostly by doing everything as `root` then `chown`ing later.

Reply:
0d07ab5